### PR TITLE
Add DefaultAzureCredential_options sample

### PR DESCRIPTION
**Pull request by baanders**
Update code screenshot in https://learn.microsoft.com/en-us/azure/digital-twins/troubleshoot-error-404-digital-twins#solution-2 to use an actual code snippet.

Accompanies this docs PR: https://github.com/MicrosoftDocs/azure-docs-pr/pull/220800

### CODE REVIEW

File: sdks/csharp/authentication.cs [hunk 1]
```diff
@@ -29,8 +29,28 @@ internal void RunSample()
...
+    // </DefaultAzureCredential_full>
+
+    // ------------------ DefaultAzureCredential with options  ------------------ 
+    // <DefaultAzureCredential_options>
+    public class DefaultAzureCredentialOptionsSample
+    {
+        // The URL of your instance, starting with the protocol (https://)
+        private const string adtInstanceUrl = "https://<your-Azure-Digital-Twins-instance-URL>";
+
+        private static DefaultAzureCredentialOptions credOpts = new DefaultAzureCredentialOptions()
```
Comment (baanders) [inline sdks/csharp/authentication.cs:42]: Changed from "var" to "private static DefaultAzureCredentialOptions" to resolve 2 build errors:

1. This line (42) was throwing "The contextual keyword 'var' may only appear within a local variable declaration or in script code"
2. Line 49 was throwing "A field initializer cannot reference the non-static field, method, or property 'DefaultAzureCredentialOptionsSample.credOpts"

File: sdks/csharp/authentication.cs [hunk 2]
```diff
@@ -29,8 +29,28 @@ internal void RunSample()
...
+        {
+            ExcludeSharedTokenCacheCredential = true,
+            ExcludeVisualStudioCodeCredential = true,
+            TenantId = "a000a000-0aaa-0000-00a0-aa000aa0aa0a"
+        };
+
+        private static DefaultAzureCredential cred = new DefaultAzureCredential(credOpts);
```
Comment (andyatwork) [inline sdks/csharp/authentication.cs:49]: Consider spelling out `cred` to `credential` for code clarity.

File: sdks/csharp/authentication.cs (hunk 1 above)
Comment (andyatwork) [inline sdks/csharp/authentication.cs:42]: Consider spelling out `credOpts` to `credentialOptions` for code readability.

File: sdks/csharp/authentication.cs [hunk 3]
```diff
@@ -29,8 +29,28 @@ internal void RunSample()
...
+    public class DefaultAzureCredentialOptionsSample
+    {
+        // The URL of your instance, starting with the protocol (https://)
+        private const string adtInstanceUrl = "https://<your-Azure-Digital-Twins-instance-URL>";
+
+        private static DefaultAzureCredentialOptions credOpts = new DefaultAzureCredentialOptions()
+        {
+            ExcludeSharedTokenCacheCredential = true,
+            ExcludeVisualStudioCodeCredential = true,
+            TenantId = "a000a000-0aaa-0000-00a0-aa000aa0aa0a"
```
Comment (andyatwork) [inline sdks/csharp/authentication.cs:46]: Should we have an angled bracket placeholder here as well to indicate that this should be replaced? E.g.: `<Azure Active Directory Tenant Id>`

**Review by andyatwork — APPROVED** [review]

**Pull request merged**